### ISSUE TITLE
MariaDB: Can't browse Packages and Sequences

### ISSUE THREAD

**Issue by zongo811** [dbeaver/dbeaver#15285]
#### System information: 
- Operating system (distribution) and version: Debian Buster
- DBeaver version: 21.2.3
- Additional extensions: None

#### Connection specification:
- Database name and version: MariaDB, 10.5.12-MariaDB-1:10.5.12+maria~buster
- Driver name: Don't know, default driver was downloaded automagically
- Do you use tunnels or proxies (SSH, SOCKS, etc)? No

#### Describe the problem you're observing:
When I click on "Packages" this error occurs:
![grafik](https://user-images.githubusercontent.com/77285389/150991588-5dc9ca1e-9ca6-4762-9706-79558172e50b.png)
Regarding this I found http://knowledge.3essentials.com/web-hosting/article/990/ERROR-SELECT-command-denied-to-user-user-x-for-table-proc.html, but have no clue how to apply it.

I also created a sequence, but this is not displayed in the Database Navigator. The sequence can be used with sql:
```
CREATE OR REPLACE SEQUENCE test_seq INCREMENT BY 1 MINVALUE 1 NOCYCLE;
select nextval(test_seq);
```

#### Steps to reproduce, if exist:
1. Create sequence and test it:
 ```
CREATE OR REPLACE SEQUENCE test_seq INCREMENT BY 1 MINVALUE 1 NOCYCLE;
select nextval(test_seq);
```
2. Use "Refresh"-Funktion of Database Navigator, then click on node "Packages" or "Sequences"

#### Include any warning/errors/backtraces from the logs

None

**Comment by Matvey16** [comment]
@zongo811 Both issues are caused by your privileges restrictions. The fact that you don't see sequences is caused by the same access restriction, but for some reason doesn't show an error.

**Comment by zongo811** [comment]
Thank you for the quick response. I know this is not DBeaver, but maybe you can help me out?
I already tried to grant privileges as User _116b3574bd7013c0:
```
MariaDB [_116b3574bd7013c0]> grant SELECT on proc to _116b3574bd7013c0;
ERROR 1142 (42000): GRANT command denied to user '_116b3574bd7013c0'@'localhost' for table 'proc'
```

And as root:
```
MariaDB [(none)]> grant SELECT on _116b3574bd7013c0.proc to '_116b3574bd7013c0'@'localhost';
ERROR 1146 (42S02): Table '_116b3574bd7013c0.proc' doesn't exist
```
As for sequences: googling and reading the mariadb-docs revealed nothing. 
Do you know by chance how to set which privileges in mariadb?

**Comment by LonwoLonwo** [comment]
I think, that the right command in your case for root is: 

```sql
GRANT Select ON mysql.proc TO '_116b3574bd7013c0'@'localhost'
```

also, can you execute this statement?

```sql
SELECT TABLE_NAME FROM information_schema.TABLES WHERE TABLE_TYPE = 'SEQUENCE'
```

We use it for the sequence reading.

**Comment by zongo811** [comment]
Hi,
I executed the grant statement as root. This fixed the problem with browsing packages.
I also executed the SELECT on TABLES. This also worked in DBeaver:
![grafik](https://user-images.githubusercontent.com/77285389/151162939-df92c326-a9b8-4651-8fb9-32b4edcf73c4.png)
Alas a Refresh on "Sequences" still did not show the sequences revealed by the SQL-statement.
I used "Invalidate/Reconnect" on the database connection, but still no luck.

I can use that. For those who are watching, here is a workaround.
For a list of sequences in your DB use
```
SELECT TABLE_NAME FROM information_schema.TABLES WHERE TABLE_TYPE = 'SEQUENCE';
```
For information about a sequence use
```
SELECT * FROM <sequence-name>;
```
or
```
SHOW CREATE SEQUENCE  <sequence-name>;
```
Thanks to all for helping me out.
